src/objective/rank_objective.hpp

```diff
@@ -134,42 +145,85 @@ class LambdarankNDCG : public RankingObjective {
     }
     // construct sigmoid table to speed up sigmoid transform
     ConstructSigmoidTable();
+
+    // initialize position bias vectors
+    InitPositionBiasesAndGradients();
+  }
+
+  void GetGradients(const double* score, score_t* gradients,
+                    score_t* hessians) const override {
+    RankingObjective::GetGradients(score, gradients, hessians);
+
+    if (unbiased_) { UpdatePositionBiasesAndGradients(); }
```

Comment:
only compute position bias vectors if `lambdarank_unbiased = true` in the config file